rest/route/annotations.go

```diff
@@ -319,3 +320,107 @@ func (h *annotationByTaskPutHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	return responder
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PUT /rest/v2/tasks/{task_id}/created_ticket
+
+// this api will be used by teams who set up their own web hook to file a ticket, to send
+// us the information for the ticket that was created so that we can store and display it.
+
+type createdTicketByTaskPutHandler struct {
+	taskId    string
+	execution *int
+	user      gimlet.User
+	ticket    *model.APIIssueLink
+	sc        data.Connector
+}
+
+func makeCreatedTicketByTask(sc data.Connector) gimlet.RouteHandler {
+	return &createdTicketByTaskPutHandler{
+		sc: sc,
+	}
+}
+
+func (h *createdTicketByTaskPutHandler) Factory() gimlet.RouteHandler {
+	return &createdTicketByTaskPutHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *createdTicketByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+	h.taskId = gimlet.GetVars(r)["task_id"]
+	if h.taskId == "" {
+		return gimlet.ErrorResponse{
+			Message:    "task ID cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	// for now, tickets will be created for a specific task execution only
+	// and will not be visible on other executions
+	if r.URL.Query().Get("execution") == "" {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("the task execution must be specified"),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	execution, err := strconv.Atoi(r.URL.Query().Get("execution"))
+	h.execution = &execution
+
+	// check if the task exists
+	t, err := task.FindOne(task.ById(h.taskId))
+	if err != nil {
+		return errors.Wrap(err, "error finding task")
+	}
+	if t == nil {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("the task '%s' does not exist", h.taskId),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	// if there is no custom web-hook configured, return an error because the
+	// purpose of this endpoint is to store the ticket created by the web-hook
+	if !graphql.IsWebhookConfigured(t) {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("there is no web-hook configured for '%s'", t.Project),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	err = json.NewDecoder(body).Decode(&h.ticket)
+	if err != nil {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("API error while unmarshalling JSON: '%s'", err.Error()),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	//validate the url
+	if err = util.CheckURL(utility.FromStringPtr(h.ticket.URL)); err != nil {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("the url is not valid: '%s' ", err.Error()),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	u := MustHaveUser(ctx)
+	h.user = u
+	return nil
+}
+
+func (h *createdTicketByTaskPutHandler) Run(ctx context.Context) gimlet.Responder {
+	// todo: error if the bb is configured
```

Comment:
Is this a todo for a future ticket or leftover?

Reply:
leftover, removed it. Thanks!